CallAutomation_OutboundCalling/src/main/java/com/communication/callautomation/ProgramSample.java

```diff
@@ -0,0 +1,194 @@
+                        log.info("Played media file with {}", playToAll(callConnectionId, "Goodbye.wav"));
+                        break;
+
+                    default:
+                        log.info("Another DTMF received. {}.", hangUp(callConnectionId));
+                        break;
+                }
+            }
+            else if(event instanceof PlayCompleted) {
+                log.info("Received Play Completed event: {}", hangUp(callConnectionId));
+            }
+            else if(event instanceof RecognizeFailed || event instanceof PlayFailed) {
+                log.error("Received failed event: {}", ((CallAutomationEventBaseWithReasonCode) event)
+                        .getResultInformation().getMessage());
+                hangUp(callConnectionId);
+            }
+        }
+        return ResponseEntity.ok().body("");
+    }
+
+    private String createOutboundCall() {
+        try {
+            PhoneNumberIdentifier caller = new PhoneNumberIdentifier(appConfig.getCallerphonenumber());
+            PhoneNumberIdentifier target = new PhoneNumberIdentifier(appConfig.getTargetphonenumber());
+            CallInvite callInvite = new CallInvite(target, caller);
+            CreateCallOptions createCallOptions = new CreateCallOptions(callInvite, appConfig.getCallBackUri());
+            Response<CreateCallResult> result = asyncClient.createCallWithResponse(createCallOptions).block();
+            return result.getValue().getCallConnectionProperties().getCallConnectionId();
+        } catch (CommunicationErrorResponseException e) {
+            log.error("Error when creating call: {} {}",
+                    e.getMessage(),
+                    e.getCause());
+            return "";
+        }
+    }
+
+    private String callRecording(final String callConnectionId) {
+        try {
+            log.info("Received Call Connected event, start call recording initiated.");
+            ServerCallLocator serverCallLocator = new ServerCallLocator(
+                    asyncClient.getCallConnectionAsync(callConnectionId)
+                            .getCallProperties().block()
+                            .getServerCallId());
+            StartRecordingOptions startRecordingOptions = new StartRecordingOptions(serverCallLocator);
+            Response<RecordingStateResult> response = asyncClient.getCallRecordingAsync()
+                    .startWithResponse(startRecordingOptions).block();
+            return response.getValue().getRecordingId();
+        } catch (Exception e) {
+            log.error("An error occurred when starting call recording: {} {}",
+                    e.getMessage(),
+                    e.getCause());
+            return "";
+        }
+    }
+
+    private String recognizeDtmfTones(final String callConnectionId, final String mediaFile) {
+        try {
+            CommunicationIdentifier target = CommunicationIdentifier
+                    .fromRawId("4:" + appConfig.getTargetphonenumber());
```

Comment:
Can you use PhoneNumberIdentifier, I believe in that case you don't need to prefix "4:" and able to use direct phone number from config

Reply:
good catch! thanks
fixed